crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -857,228 +857,258 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
+                                if router_data.status == enums::AttemptStatus::Charged {
+                                    Some(None)
+                                } else {
+                                    None
+                                };
+
+                            if router_data.status == enums::AttemptStatus::Charged {
+                                payment_data
+                                    .payment_intent
+                                    .fingerprint_id
+                                    .clone_from(&payment_data.payment_attempt.fingerprint_id);
+                                metrics::SUCCESSFUL_PAYMENT.add(&metrics::CONTEXT, 1, &[]);
+                            }
 
-                    let payment_method_id = payment_data.payment_attempt.payment_method_id.clone();
+                            let payment_method_id =
+                                payment_data.payment_attempt.payment_method_id.clone();
+
+                            utils::add_apple_pay_payment_status_metrics(
+                                router_data.status,
+                                router_data.apple_pay_flow.clone(),
+                                payment_data.payment_attempt.connector.clone(),
+                                payment_data.payment_attempt.merchant_id.clone(),
+                            );
+                            let (capture_updates, payment_attempt_update) = match payment_data
+                                .multiple_capture_data
+                            {
+                                Some(multiple_capture_data) => {
+                                    let capture_update = storage::CaptureUpdate::ResponseUpdate {
+                                        status: enums::CaptureStatus::foreign_try_from(
+                                            router_data.status,
+                                        )?,
+                                        connector_capture_id: connector_transaction_id.clone(),
+                                        connector_response_reference_id,
+                                    };
+                                    let capture_update_list = vec![(
+                                        multiple_capture_data.get_latest_capture().clone(),
+                                        capture_update,
+                                    )];
+                                    (Some((multiple_capture_data, capture_update_list)), None)
+                                }
+                                None => (
+                                    None,
+                                    Some(storage::PaymentAttemptUpdate::ResponseUpdate {
+                                        status: updated_attempt_status,
+                                        connector: None,
+                                        connector_transaction_id: connector_transaction_id.clone(),
+                                        authentication_type: None,
```

Comment:
why is this update made as None?

Reply:
had removed it while resolving merge conflicts my bad! Adding it back